core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -89,4 +89,11 @@ package object config {
     .stringConf
     .toSequence
     .createWithDefault(Nil)
+
+  // Note: This is a SQL config but needs to be in core because the REPL depends on it
+  private[spark] val CATALOG_IMPLEMENTATION = ConfigBuilder("spark.sql.catalogImplementation")
+    .internal()
+    .stringConf
+    .checkValues(Set("hive", "in-memory"))
```

Comment:
we use this for both `SharedState` and `SessionState`. If we want to use a class name then we need two configs.